core/src/main/scala/kafka/log/LogValidator.scala

```diff
@@ -186,7 +186,7 @@ private[kafka] object LogValidator {
       if (record.magic != messageFormatVersion)
         inPlaceAssignment = false
 
-      validatedRecords += record.convert(messageFormatVersion)
+      validatedRecords += record.convert(messageFormatVersion, messageTimestampType)
```

Comment:
Good question. It seems it may actually be possible for the user to use "NO_TIMESTAMP_TYPE" for "message.timestamp.type." If so, it's probably worth checking for that and raising an exception.